File: backend/recouvrex-services/src/main/java/com/recouvrex/process/model/guarantee/VehicleGuarantee.java
package com.recouvrex.process.model.guarantee;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "vehicle_guarantee") // Nom de la table dans la base de données
// Cette classe représente une garantie de véhicule, étendant la classe de base Garantie.
public class VehicleGuarantee  extends Guarantee {

    // The brand of the vehicle.
    private String vehicleBrand;

    // The model year of the vehicle.
    private Integer modelYear;

    // The registration number of the vehicle.
    private String registrationNumber;

    // The type of fuel used by the vehicle.
    private String fuelType;

    // The fiscal horsepower of the vehicle.
    private Integer fiscalHorsepower;

}

File: frontend/src/models/guarantee/vehicleGuarantee.ts
// vehicleGuarantee.ts

import { Credit } from "../Credit";


export interface VehicleGuarantee {
  id: number;
  credit: Pick<Credit, 'id'>;
  vehicleBrand: string;
  modelYear: number;
  registrationNumber: string;
  fuelType: string;
  fiscalHorsepower: number;
  type: string;
}
